### PR TITLE
LOG-3613: Bump k8s client go version to 1.24.9

### PR DESCRIPTION
**Pull request by vparfonov**
### Description
This PR: 

- bump `k8s client-go` version to `1.24.9` from `1.23.5`
- bump `prometheus-operator` to `v0.55.1` from `v0.38.1 `
- remove `replace k8s.io/client-go => k8s.io/client-go v0.23.5 // Required by prometheus-operator` from go.mod
- `kubeclock.Clock` is deprecated: Use `k8s.io/utils/clock.WithTickerAndDelayedExecution` instead

/cc 
/assign @jcantrill 

/cherry-pick 

### Links

- Depending on PR(s): https://github.com/openshift/eventrouter/pull/40
- Bugzilla:
- Github issue:
- JIRA: https://issues.redhat.com/browse/LOG-3614
- Enhancement proposal:

### CODE REVIEW

File: internal/utils/comparators/servicemonitor/comparator.go [hunk 1]
```diff
@@ -4,72 +4,72 @@ import (
...
+  }
 
-	if !utils.AreMapsSame(current.ObjectMeta.Labels, desired.ObjectMeta.Labels) {
-		log.V(3).Info("ServiceMonitor label change", "current name", current.Name)
-		return false
-	}
+  if !utils.AreMapsSame(current.ObjectMeta.Labels, desired.ObjectMeta.Labels) {
+    log.V(3).Info("ServiceMonitor label change", "current name", current.Name)
+    return false
+  }
```
Comment (vimalk78) [inline internal/utils/comparators/servicemonitor/comparator.go:23]: is this change done after running `go fmt` ?  the Go standard is to use tabs

Comment (vimalk78) [reply internal/utils/comparators/servicemonitor/comparator.go:23]: same comment for other files

**Comment by vimalk78** [comment]
Switch to the "rich diff" to see file diff, and it shows a Critical Severity issue in one of the dependencies in go.mod
https://github.com/openshift/cluster-logging-operator/pull/1885/files?diff=unified&w=0&short_path=33ef32b#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

File: internal/utils/comparators/servicemonitor/comparator.go (hunk 1 above)
Comment (vparfonov) [reply internal/utils/comparators/servicemonitor/comparator.go]: hm, strange i run `make fmt`

**Comment by jcantrill** [comment]
> Switch to the "rich diff" to see file diff, and it shows a Critical Severity issue in one of the dependencies in go.mod https://github.com/openshift/cluster-logging-operator/pull/1885/files?diff=unified&w=0&short_path=33ef32b#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

@vparfonov we might as well patch these critical libs with this change

**Comment by vparfonov** [comment]
@vimalk78 @jcantrill fixed your comments

**Pull request merged**